src/linker/Linker.Steps/RemoveUnreachableBlocksStep.cs

```diff
@@ -313,8 +313,28 @@ public bool RewriteBody ()
 
 				bodySweeper.Process (conditionInstrsToRemove);
 				InstructionsReplaced = bodySweeper.InstructionsReplaced;
+				if (InstructionsReplaced == 0)
+					return false;
+
+				reachableInstrs = GetReachableInstructionsMap (out _);
+				if (reachableInstrs != null)
+					RemoveUnreachableInstructions (reachableInstrs);
+
+				return true;
+			}
+
+			void RemoveUnreachableInstructions (BitArray reachable)
+			{
+				ILProcessor processor = Body.GetILProcessor ();
+
+				int removed = 0;
+				for (int i = 0; i < reachable.Count; ++i) {
+					if (reachable [i])
+						continue;
 
-				return InstructionsReplaced > 0;
+					processor.RemoveAt (i - removed);
```

Comment:
Do you have logic to update the operand of branch instructions and Start/End/etc on ExceptionHandler?

Reply:
That's automatically updated by cecil